src/dataflow/src/render/reduce.rs

```diff
@@ -502,12 +508,66 @@ impl ReducePlan {
                     ));
                 }
                 // Now we need to collate them together.
-                build_collation(to_collate, expr.aggregate_types, &mut collection.scope())
+                build_collation(to_collate, expr.aggregate_types, &mut collection.scope()).into()
             }
+        };
+        arrangement_or_bundle.to_bundle(key_arity, err_input)
+    }
+}
+
+enum ArrangementOrBundle<G, T>
+where
+    G: Scope,
+    G::Timestamp: Lattice + Refines<T>,
+    T: Timestamp + Lattice,
+{
+    Arrangement(Arrangement<G, Row>),
+    CollectionBundle(CollectionBundle<G, Row, T>),
+}
```

Comment:
Can you talk me through this enum? A `CollectionBundle` can represent a single arrangement, though I suppose it has to have errors with it?

Reply:
If this is just meant to be a minor local enum for the purposes of simplifying some reduce rendering, that's no worries, but I would put some doc comments on it explaining what the goal is (e.g. why not make it public and ship it to others, that the two variants reflect errors in one case and not the other, idk stuff like that.

Reply:
That's a good point, the current type could be confusing. A `CollectionBundle` itself might contain an arrangement. What I really intended here was to give reductions a choice to either return an arrangement or two collections. I'll see if I can change it to be clearer.

Reply:
Pondering a bit: if it is just to make the `match` for reduce rendering easier, no worries but maybe doc as such. If it is meant to be a re-usable abstraction, we should workshop further.

Reply:
I changed the type to contain a single `Collection` instead of a `CollectionBundle`. This a) doesn't require it to mess with the `err_collection`, and b) it's leaner and easier to understand.